tests/unit/test_discovery_client.py

```diff
@@ -79,6 +91,81 @@ def test___discovery_service_unavailable___register_service_registration_failure
     assert ~unregistration_success_flag  # False
 
 
+def test___get_discovery_service_address___start_service_jit___returns_expected_value(
+    mocker: MockerFixture,
+    temp_discovery_key_file_path: pathlib.Path,
+    temp_registration_json_file_path: pathlib.Path,
+    discovery_service_temp_directory: pathlib.Path,
+):
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._get_key_file_path",
+        return_value=temp_discovery_key_file_path,
+    )
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._service_already_running",
+        return_value=False,
+    )
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._get_registration_json_file_path",
+        return_value=temp_registration_json_file_path,
+    )
+    mock_popen = mocker.patch("subprocess.Popen")
+
+    discovery_service_address = _get_discovery_service_address()
+
+    for call in mock_popen.call_args_list:
+        args, _ = call
+        (mock_file_path_arg,) = args
+        assert (
+            mock_file_path_arg[0]
+            == discovery_service_temp_directory
+            / json.loads(_MOCK_REGISTRATION_FILE_CONTENT)["discovery"]["path"]
+        )
+
+    assert mock_popen.call_count == 1
```

Comment:
simplify
```suggestion
    exe_file_path = (discovery_service_temp_directory / json.loads(_MOCK_REGISTRATION_FILE_CONTENT)["discovery"]["path"])
    mock_popen.assert_called_once_with([exe_file_path], cwd=exe_file_path.parent)
```

You can also use `mock_popen.call_args.args[0]` to inspect the 1st parameter for the most recent call.

Reply:
Replaced with a simplified code